File: app/src/main/java/com/xando/chefsclub/Search/Core/DataToItemsSetter.java
package com.xando.chefsclub.Search.Core;

import android.support.annotation.NonNull;

import com.mikepenz.fastadapter.items.AbstractItem;
import com.xando.chefsclub.DataWorkers.BaseData;

import java.util.List;

public interface DataToItemsSetter<Data extends BaseData, Item extends AbstractItem & IData<Data>> {
    @NonNull
    Item[] getItems(@NonNull List<Data> dataList);
}

File: app/src/main/java/com/xando/chefsclub/Search/Core/DataToItemsSetter.kt
package com.xando.chefsclub.Search.Core

import android.support.v7.widget.RecyclerView
import com.mikepenz.fastadapter.IItem
import com.xando.chefsclub.DataWorkers.BaseData

interface DataToItemsSetter<Data : BaseData, Item>
        where Item : IItem<out Any, out RecyclerView.ViewHolder>, Item : IData<Data> {
    fun getItems(dataList: List<Data>): List<Item>
}